### PR TITLE
feat(verification): improve transient error handling

### PR DESCRIPTION
**Pull request by hiddeco**
Addresses (part of) #1640

This improves the handling of transient Kubernetes API errors which may occur during the verification step of a Stage's current Freight.

It does this by selectively returning errors which should be retried, causing the reconciler to not leave the `Verifying` phase and allowing it to retry the action it performed on the next reconciliation attempt.

To allow the reconciler to distinguish a creation error from a failed attempt to retrieve results, closer inspection is performed on the existence of an `AnalysisRun` reference in the `VerificationInfo`.

### CODE REVIEW

**Comment by krancour** [comment]
This LGTM after merge conflicts are resolved.

I may be PR'ing something similar for Promotions shortly.

**Comment by hiddeco** [comment]
Has been rebased.

**Review by krancour — APPROVED** [review]

**Pull request merged**
